src/Microsoft.Azure.WebJobs.Host/Scale/DefaultConcurrencyThrottleManager.cs

```diff
@@ -66,7 +66,7 @@ public ConcurrencyThrottleAggregateStatus GetStatus()
 
         private void UpdateThrottleState()
         {
-            IEnumerable<ConcurrencyThrottleStatus> throttleResults = _throttleProviders.Select(p => p.GetStatus(_logger));
+            IEnumerable<ConcurrencyThrottleStatus> throttleResults = _throttleProviders.Select(p => p.GetStatus(_logger)).ToArray();
```

Comment:
E2E testing showed that there was a subtle bug here due to the fact that this enumerable wasn't cached. Because of this, I could get inconsistent/flaky results below.